src/Raven.Server/Documents/Patch/ScriptRunner.cs

```diff
@@ -528,6 +533,87 @@ private JsValue AppendTimeSeries(JsValue document, JsValue name, JsValue[] args)
                 return Undefined.Instance;
             }
 
+            private JsValue IncrementTimeSeries(JsValue document, JsValue name, JsValue[] args)
+            {
+                AssertValidDatabaseContext("timeseries(doc, name).increment");
+
+                const string signature2Args = "timeseries(doc, name).increment(timestamp, values)";
+
+                if(args.Length != 2)
+                    throw new ArgumentException($"There is no overload with {args.Length} arguments for this method should be {signature2Args}");
+
+                string signature = signature2Args;
+
+                var (id, doc) = GetIdAndDocFromArg(document, _timeSeriesSignature);
+
+                string timeSeries = GetStringArg(name, _timeSeriesSignature, "name");
+                var timestamp = GetTimeSeriesDateArg(args[0], signature, "timestamp");
+
+                double[] valuesBuffer = null;
+                try
+                {
+                    var valuesArg = args[1];
+                    Memory<double> values;
+                    if (valuesArg.IsArray())
+                    {
+                        var jsValues = valuesArg.AsArray();
+                        valuesBuffer = ArrayPool<double>.Shared.Rent((int)jsValues.Length);
+                        FillDoubleArrayFromJsArray(valuesBuffer, jsValues, signature);
+                        values = new Memory<double>(valuesBuffer, 0, (int)jsValues.Length);
+                    }
+                    else if (valuesArg.IsNumber())
+                    {
+                        valuesBuffer = ArrayPool<double>.Shared.Rent(1);
+                        valuesBuffer[0] = valuesArg.AsNumber();
+                        values = new Memory<double>(valuesBuffer, 0, 1);
+                    }
+                    else
+                    {
+                        throw new ArgumentException($"{signature}: The values should be an array but got {GetTypes(valuesArg)}");
+                    }
+
+                    var tss = _database.DocumentsStorage.TimeSeriesStorage;
+                    var newSeries = tss.Stats.GetStats(_docsCtx, id, timeSeries).Count == 0;
+
+                    if (newSeries)
+                    {
+                        DocumentTimeSeriesToUpdate ??= new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+                        DocumentTimeSeriesToUpdate.Add(id);
+                    }
+
+                    var toIncrement = new TimeSeriesOperation.IncrementOperation
+                    {
+                        Values = values.ToArray(),
```

Comment:
You are creating an allocation here, after we were careful to avoid it using array pool

Reply:
I created a new issue for TS allocations (https://issues.hibernatingrhinos.com/issue/RavenDB-17729) - we'll fix it there :-)